flytekit/extras/tasks/shell.py

```diff
@@ -237,3 +256,32 @@ def execute(self, **kwargs) -> typing.Any:
 
     def post_execute(self, user_params: ExecutionParameters, rval: typing.Any) -> typing.Any:
         return self._config_task_instance.post_execute(user_params, rval)
+
+
+# The portable_shell_task is an instance of ShellTask which wraps a 'pure' shell script
+# This utility function allows for the specification of env variables, arguments, and the actual script within the
+# workflow definition rather than at `ShellTask` instantiation
+def get_portable_shell_task() -> ShellTask:
+    return ShellTask(
+    name="portable_shell_task_instance",
+    debug=True,
+    inputs=flytekit.kwtypes(env=typing.Dict[str, str], script_args=str, script_file=str),
+    output_locs=[
+        OutputLocation(
+            var="k",
```

Comment:
it is also assuming that k is always the output?

Reply:
I wasn't sure how best to handle this, I think most standalone shell scripts either write files to some known location or accept an output file/dir as input parameter. I could have no outputs but in the spirit of returning some output to be passed to a downstream task, I just return the `ctx.working_directory`. I suppose the variable name could be anything here but I used `k` as in the examples